### PR TITLE
fix: return unique_list only when link_extractor.unique is True

### PR DESCRIPTION
**Pull request by reidemeister94**
In the `lxmlhtml.py` script, the function `extract_links(self, response)` always returns `unique_list(all_links)`, even when the attribute `unique` of the `link_extractor` is equal to `False`. 

The behavior is coherent with the `unique` attribute with this little change in the return line.

Fixes https://github.com/scrapy/scrapy/issues/3798, closes https://github.com/scrapy/scrapy/pull/3799, closes https://github.com/scrapy/scrapy/pull/4695

### CODE REVIEW

**Comment by wRAR** [comment]
See also: #3798 #4695 #3799

**Comment by reidemeister94** [comment]
Ok so this won't get fixed?

**Comment by reidemeister94** [comment]
Any update on this?

**Comment by Gallaecio** [comment]
Added to my review queue, but it may take a while :bow:

**Comment by reidemeister94** [comment]
Just a ping :D

**Comment by Gallaecio** [comment]
I am OK with merging as is, provided we update the docstring of `extract_links` and provide a test for the change.

**Comment by reidemeister94** [comment]
I updated the docstring and fixed the html used for the tests to have tests that also cover the new case (unique = false, and the output contains duplicates). 
I hope that's enough for being merged :D

**Review by Gallaecio — APPROVED** [review]
Until your tests changes I did not realize the issue only affected links with matching URL *and* text, but the original issue indeed describes it like that.

Thanks!

File: scrapy/linkextractors/lxmlhtml.py [hunk 1]
```diff
@@ -195,7 +195,8 @@ def extract_links(self, response):
         Only links that match the settings passed to the ``__init__`` method of
         the link extractor are returned.
 
-        Duplicate links are omitted.
+        Duplicate links are omitted if the ``unique`` parameter is set to ``True``,
```
Comment (Gallaecio) [inline scrapy/linkextractors/lxmlhtml.py:198]:
```suggestion
        Duplicate links are omitted if the ``unique`` attribute is set to ``True``, 
```

**Review by wRAR — APPROVED** [review]

**Pull request merged**